src/site/content/en/blog/streams/index.md

```diff
@@ -0,0 +1,966 @@
+method of the
+[`WritableStreamDefaultWriter`](https://developer.mozilla.org/en-US/docs/Web/API/WritableStreamDefaultWriter)
+interface writes a passed chunk of data to a `WritableStream` and its underlying sink, then returns
+a promise that resolves to indicate the success or failure of the write operation. Note that what
+"success" means is up to the underlying sink; it might indicate that the chunk has been accepted,
+and not necessarily that it is safely saved to its ultimate destination.
+
+#### The `locked` property
+
+You can check if a writable stream is locked by accessing its
+[`WritableStream.locked`](https://developer.mozilla.org/en-US/docs/Web/API/WritableStream/locked)
+property.
+
+### Writable stream code sample
+
+The code sample below shows all steps in action.
+
+```js
+const writableStream = new WritableStream({
+  async start(controller) {
+    console.log('[start]');
+  },
+  async write(chunk, controller) {
+    console.log('[write]', chunk);
+    // Wait for next write.
+    await new Promise((resolve) => setTimeout(resolve, 1000));
+  },
+  async close(controller) {
+    console.log('[close]');
+  },
+  async abort(reason) {
+    console.log('[abort]', reason);
+  },
+});
+
+(async () => {
+  const writer = writableStream.getWriter();
+  const start = Date.now();
+  for (const char of 'abcd') {
+    // Wait to add write queue.
+    await writer.ready;
+    console.log('[ready]', Date.now() - start, 'ms');
+    // The Promise is resolved after the write finishes.
+    writer.write(char);
+  }
+  await writer.close();
+})();
+```
+
+### Piping a readable stream to a writable stream
+
+A readable stream can be piped to a writable stream through the readable stream's
+[`pipeTo()`](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream/pipeTo) method.
+`ReadableStream.pipeTo()` pipes the current `ReadableStream`to a given`WritableStream` and returns a
+promise that fulfills when the piping process completes successfully, or rejects if any errors were
+encountered.
+
+```js
+const readableStream = new ReadableStream({
+  async start(controller) {
+    // Called by constructor.
+    console.log('[start readable]');
+    controller.enqueue('a');
+    controller.enqueue('b');
+    controller.enqueue('c');
+  },
+  async pull(controller) {
+    // Called when controller's queue is empty.
+    console.log('[pull]');
+    controller.enqueue('d');
+    controller.close();
+  },
+  async cancel(reason) {
+    // Called when the stream is canceled.
+    console.log('[cancel]', reason);
+  },
+});
+
+const writableStream = new WritableStream({
+  async start(controller) {
+    // Called by constructor
+    console.log('[start writable]');
+  },
+  async write(chunk, controller) {
+    // Called when writer.write()
+    console.log('[write]', chunk);
+  },
+  async close(controller) {
+    console.log('[close]');
+  },
+  async abort(reason) {
+    console.log('[abort]', reason);
+  },
+});
+
+(async () => {
```

Comment:
Once again, it's not needed there.
A comment though would be appreciated such as the console output.

Reply:
I didn't expect `pull` for instance. Explaining it would be beneficial to users.
```
 [start readable]
 [start writable]
 [write] a
 [write] b
 [pull]
 [write] c
 [write] d
 [close]
 [finished]
```